volttron/utils/rmq_setup.py

```diff
@@ -168,7 +168,48 @@ def write_env_file(rmq_config, conf_file, env=None):
             env_conf.write(env_entries)
 
 
-def _create_federation_setup(admin_user, admin_password, is_ssl, vhost, vhome):
+def _get_federation_certs(vhome):
+    federation_config_file = os.path.join(vhome,
+                                          'rabbitmq_federation_config.yml')
+    federation_config = read_config_file(federation_config_file)
+    federation = federation_config.get('federation-upstream', {})
+    success = True
+    update_needed = False
+    try:
+        for host, upstream in federation.items():
+            rmq_user = 'federation' #upstream['federation-user']
+            if 'certificates' not in upstream:
+                # certificates key not found in shovel config
+                https_port = upstream.get('https_port', 8443)
+                remote_addr = 'https://{}:{}'.format(host, https_port)
+                print(f"Certificates not found. Requesting CSR from {remote_addr}")
+                # request CSR from remote host
+                ca_file, cert_file, prvt_file = _request_csr(rmq_user, remote_addr, 'federation')
+                if ca_file is not None and cert_file is not None and prvt_file is not None:
+                    upstream['certificates'] = {}
+                    # root CA
+                    upstream['certificates']['remote_ca'] = ca_file
+                    # public cert
+                    upstream['certificates']['public_cert'] = cert_file
+                    # private_key
+                    upstream['certificates']['private_key'] = prvt_file
+                    update_needed = True
+                else:
+                    _log.error(
+                        f"ERROR: Couldn't get CSR certificates from remote server. Check with admin of remote instance."
+                        "\nContinuing with other configurations")
+                    continue
```

Comment:
Also the error message here is different from the error message I get  when I do vcfg rabbitmq federation without a config file. Error message should be consistent. When no config file is used and csr is denied,  the error message clearly says that the csr was denied. I should get the same message when config file is used

When max retry is reached while using config file, I see a stack trace before the message "max retry attempt reached". When the same happens when running without config file there is no stack trace but just the message is displayed.  I like the latter 

**
Your VOLTTRON_HOME currently set to: /home/volttron/fed_v1

Is this the volttron you are attempting to setup? [Y]:
Name of this volttron instance: [volttron1]:
rabbitmq_federation_config.yml exists in /home/volttron/fed_v1/rabbitmq_federation_config.yml Do you wish to use this file to configure federation [Y]:
Certificates not found. Requesting CSR from https://volttron2:8443
INFO: Waiting for administrator to accept the CSR request.
Traceback (most recent call last):
  File "/home/volttron/git/myvolttron/env/bin/vcfg", line 11, in <module>
    load_entry_point('volttron', 'console_scripts', 'vcfg')()
  File "/home/volttron/git/myvolttron/volttron/platform/config.py", line 588, in _main
    main()
  File "/home/volttron/git/myvolttron/volttron/platform/instance_setup.py", line 1029, in main
    process_rmq_inputs(vars(args))
  File "/home/volttron/git/myvolttron/volttron/platform/instance_setup.py", line 967, in process_rmq_inputs
    setup_rabbitmq_volttron(args_dict['installation-type'], verbose, prompt=True, instance_name=instance_name, max_retries=args_dict['max_retries'])
  File "/home/volttron/git/myvolttron/volttron/utils/rmq_setup.py", line 748, in setup_rabbitmq_volttron
    success = _get_federation_certs(rmq_config.volttron_home)
  File "/home/volttron/git/myvolttron/volttron/utils/rmq_setup.py", line 187, in _get_federation_certs
    ca_file, cert_file, prvt_file = _request_csr(rmq_user, remote_addr, 'federation')
  File "/home/volttron/git/myvolttron/volttron/utils/rmq_setup.py", line 1222, in _request_csr
    raise TimeoutError("Maximum retry attempts for CSR reached. "
TimeoutError: Maximum retry attempts for CSR reached. Please retry command once administrator of remote VOLTTRON instance is ready to approve certificate signing request
(volttron) volttron@volttron1:~/git/myvolttron$ vcfg rabbitmq federation --max-retries 1

Your VOLTTRON_HOME currently set to: /home/volttron/fed_v1

Is this the volttron you are attempting to setup? [Y]:
Name of this volttron instance: [volttron1]:
rabbitmq_federation_config.yml exists in /home/volttron/fed_v1/rabbitmq_federation_config.yml Do you wish to use this file to configure federation [Y]: n
INFO: New input data will be used to overwrite existing /home/volttron/fed_v1/rabbitmq_federation_config.yml
Number of upstream servers to configure: [1]:
Hostname of the upstream server:  volttron2
Port of the upstream server:  [5671]:
Virtual host of the upstream server:  [volttron]:

Do you have certificates signed by remote CA?  [N]:
Path to remote web interface:  [https://volttron2:8443]:
INFO: Waiting for administrator to accept the CSR request.
ERROR: Maximum retry attempts for CSR reached. Please retry command once administrator of remote VOLTTRON instance is ready to approve certificate signing request
(volttron) volttron@volttron1:~/git/myvolttron$